### PR TITLE
CSS parser

### PR DESCRIPTION
**Pull request by Arnau478**
This PR introduces a CSS parser, as discussed in #4

# State
The current state of the parser is the following:
- [x] Core CSS grammar
- [x] Formatter
- [x] Simple selectors (class, id, etc.)
- [x] Pseudo-classes and pseudo-elements 
- [x] Multi-expression declarations (e.g. `padding: 4px 2px;`)
- [x] Functions (e.g. `rgb(105,12,52)`)
- [x] `@media`
- [ ] `@import`, `@font-face`, `@keyframes`, etc.
- [ ] Selector combinators (`>`, `+`, etc.)
- [ ] Attribute selectors (e.g. `a[href="..."]`)
- [ ] Full parse error coverage
- [ ] Crash stability (still has many `@panic("TODO")`)

**TLDR**: It can handle a decent amount of real-world stylesheets.

Overall, I would say it's advanced enough to have it in the main repo (although not wired in yet). Also, the fact that I'm opening a PR doesn't mean that it has to be merged right away, it means that from my side it's ready. Feel free to wait a bit before merging if you feel like doing so. I'll continue to work on my own fork until then.

### Note
I'm not sure about what branch to target. GitHub lets me change it after opening the PR, so I've left it as the default (`main`) for now.

### CODE REVIEW

**Comment by kristoff-it** [comment]
I've invited you to the repo, when you accept the invite please merge your PR as your first official act :^) I would recommend creating a merge commit in this case.
